packages/graphql/src/schema-model/entity/model-adapters/ImplementingEntityOperations.ts

```diff
@@ -0,0 +1,213 @@
+/*
+ * Copyright (c) "Neo4j"
+ * Neo4j Sweden AB [http://neo4j.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { upperFirst } from "../../../utils/upper-first";
+import type { InterfaceEntityAdapter } from "./InterfaceEntityAdapter";
+import type { ConcreteEntityAdapter } from "./ConcreteEntityAdapter";
+
+export type RootTypeFieldNames = {
+    create: string;
+    read: string;
+    update: string;
+    delete: string;
+    aggregate: string;
+};
+
+type AggregateTypeNames = {
+    selection: string;
+    input: string;
+};
+
+type MutationResponseTypeNames = {
+    create: string;
+    update: string;
+};
+
+type SubscriptionEvents = {
+    create: string;
+    update: string;
+    delete: string;
+    create_relationship: string;
+    delete_relationship: string;
+};
+
+export type UpdateMutationArgumentNames = {
+    connect: string;
+    disconnect: string;
+    create: string;
+    update: string;
+    delete: string;
+    connectOrCreate: string;
+    where: string;
+};
+
+export type CreateMutationArgumentNames = {
+    input: string;
+};
+
+export class ImplementingEntityOperations<T extends InterfaceEntityAdapter | ConcreteEntityAdapter> {
+    protected readonly pascalCasePlural: string;
+    protected readonly pascalCaseSingular: string;
+
+    constructor(protected readonly entityAdapter: T) {
```

Comment:
For consistency, we try to avoid having constructor assignments

```suggestion
    protected readonly entityAdapter: T

    constructor(entityAdapter: T) {
        this.entityAdapter = entityAdapter
```

I'm aware in this case it makes the code more verbose, but it is easier to keep this consistent where possible